src/components/Envelope.vue

```diff
@@ -483,6 +475,30 @@ export default {
 				id: this.data.databaseId,
 			})
 		},
+		hasSelectAcl() {
+			if (!this.data.myAcls) {
+				return true
+			}
+			return this.data.myAcls.indexOf('r') !== -1
+		},
+		hasSeenAcl() {
+			if (!this.data.myAcls) {
+				return true
+			}
+			return this.data.myAcls.indexOf('s') !== -1
+		},
+		hasArchiveAcl() {
+			if (!this.data.myAcls) {
+				return true
+			}
+			return this.data.myAcls.indexOf('w') !== -1
```

Comment:
I cannot find where did i see that archive was index of W. so need to double check if that is correct.

Reply:
I think archiving is a bit more complicated than that because it involves moving an envelope from one mailbox to another one.

We need to check if:
1. The source/current mailbox has `t` (and probably `e`).
2. The target/archive mailbox has `i`.